tests/unit/metadata/test_table.py

```diff
@@ -621,6 +621,40 @@ def test__transform_constraints_is_condition_false_returns_data(self):
         assert result.equals(expected_result)
         assert table_instance._constraints_to_reverse == []
 
+    def test_reverse_transform_integer_rounding(self):
+        """Test the ``Table.reverse_transform`` method correctly rounds.
+
+        Expect the data to be rounded when the ``_fields_metadata`` specifies
+        the ``'transformer'`` as ``'integer'``.
+
+        Input:
+        - A dictionary with float values.
+        Output:
+        - The input dictionary rounded.
+        """
+        # Setup
+        data = pd.DataFrame({'bar': [0.2, 1.7, 2]})
+        table = Table()
+        table.fitted = True
+        table._fields_metadata = {
+            'bar': {
+                'type': 'categorical',
+                'transformer': 'integer'
+            },
+        }
+        table._hyper_transformer = Mock()
+        table._hyper_transformer.reverse_transform.return_value = data
+        table._constraints_to_reverse = []
+        table._dtypes = {'bar': 'int'}
+        table._field_names = ['bar']
+
+        # Run
+        output = table.reverse_transform(data)
+
+        # Assert
+        expected_data = pd.DataFrame({'bar': [0, 2, 2]})
+        pd.testing.assert_frame_equal(output, expected_data, check_dtype=False)
```

Comment:
why was this failing with the dtype? Shouldn't it be converted back into an int?

Reply:
Because Windows has `int32` instead of `int64`.